REGRESSION (262629@main): [ wk2 ] webgl/webgl-metal-enabled.html is a constant failure

kkinnunen-apple opened the pull request:
#### 140a13b384e73ce2e33cde7cbf8a58076f11ab52
<pre>
REGRESSION (262629@main): [ wk2 ] webgl/webgl-metal-enabled.html is a constant failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=256841">https://bugs.webkit.org/show_bug.cgi?id=256841</a>
rdar://108629326

Reviewed by Dan Glastonbury.

Remove webgl-metal-disabled.html webgl-metal-enabled.html
It&apos;s overly verbose, needing quite low-level access in Internals.
Also the condition platformSupportsMetal() is not accurate anymore
and it cannot work, as the feature detection cannot be done in WP
with IOKit blocking.

This is already tested with webgl/webgl-via-metal-flag-{off,on}.html
These test uses translated shader source to detect &quot;did it use Metal&quot;.
webgl/webgl-via-metal-flag-off.html should pass when platform supports
OpenGL.
webgl/webgl-via-metal-flag-on.html should pass when platform supports
Metal.

Add a new test:
webgl/webgl-backend-type.html

Outputs the default backend type. This can be used to ensure that the
default does not flip accidentally.

Removes Internals::hasLowAndHighPowerGPUs(), similarly current
implementation cannot work with IOKit blocking, and the function is not
used at all.

* LayoutTests/TestExpectations:
* LayoutTests/webgl/webgl-metal-disabled-expected.txt: Removed.
* LayoutTests/webgl/webgl-metal-disabled.html: Removed.
* LayoutTests/webgl/webgl-metal-enabled-expected.txt: Removed.
* LayoutTests/webgl/webgl-metal-enabled.html: Removed.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::requestedMetal): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.mm:
(WebCore::Internals::platformSupportsMetal): Deleted.

Canonical link: <a href="https://commits.webkit.org/264312@main">https://commits.webkit.org/264312@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/05874d792cc2b57dae5fb801cc6dc6d5d930e246

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7144 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7392 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7570 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8761 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7365 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7154 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8668 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7324 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10267 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7271 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/7939 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6581 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8867 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/5289 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6500 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14242 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/6949 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6600 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/9476 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7080 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5788 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6440 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/6409 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1734 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10639 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6823 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/WebCore/testing/Internals.mm:
```diff
@@ -215,27 +212,6 @@ - (NSAttributedString *)_attributedStringForRange:(NSRange)range
 }
 
-#if ENABLE(WEBGL) && PLATFORM(COCOA)
-bool Internals::platformSupportsMetal(bool isWebGL2)
```
Also need to remove `Internals::platformSupportsMetal` from `testing/Internals.cpp`

djg reviewed (approved):
:shipit:

kkinnunen-apple committed:
REGRESSION (262629@main): [ wk2 ] webgl/webgl-metal-enabled.html is a constant failure
https://bugs.webkit.org/show_bug.cgi?id=256841
rdar://108629326

Reviewed by Dan Glastonbury.

Remove webgl-metal-disabled.html webgl-metal-enabled.html
It's overly verbose, needing quite low-level access in Internals.
Also the condition platformSupportsMetal() is not accurate anymore
and it cannot work, as the feature detection cannot be done in WP
with IOKit blocking.

This is already tested with webgl/webgl-via-metal-flag-{off,on}.html
These test uses translated shader source to detect "did it use Metal".
webgl/webgl-via-metal-flag-off.html should pass when platform supports
OpenGL.
webgl/webgl-via-metal-flag-on.html should pass when platform supports
Metal.

Add a new test:
webgl/webgl-backend-type.html

Outputs the default backend type. This can be used to ensure that the
default does not flip accidentally.

Removes Internals::hasLowAndHighPowerGPUs(), similarly current
implementation cannot work with IOKit blocking, and the function is not
used at all.

* LayoutTests/TestExpectations:
* LayoutTests/webgl/webgl-metal-disabled-expected.txt: Removed.
* LayoutTests/webgl/webgl-metal-disabled.html: Removed.
* LayoutTests/webgl/webgl-metal-enabled-expected.txt: Removed.
* LayoutTests/webgl/webgl-metal-enabled.html: Removed.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::requestedMetal): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.mm:
(WebCore::Internals::platformSupportsMetal): Deleted.

Canonical link: https://commits.webkit.org/264312@main
